dot/telemetry/mailer.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package telemetry
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/gorilla/websocket"
+)
+
+var (
+	ErrInsufficientConnections = errors.New("not able to connect to any telemetry endpoints")
+	ErrTimoutMessageSending    = errors.New("timeout sending telemetry message")
+)
+var messageQueue chan Message = make(chan Message, 256)
+
+type telemetryMessage struct {
+	MessageType string    `json:"msg"`
+	Timestamp   time.Time `json:"ts"`
+	Message
+}
+
+type telemetryConnection struct {
+	wsconn    *websocket.Conn
+	verbosity int
+	sync.Mutex
+}
+
+// Handler struct for holding telemetry related things
+type mailer struct {
+	messageQueue chan Message
+	connections  []*telemetryConnection
+	logger       log.LeveledLogger
+}
+
+func newMailer(logger log.LeveledLogger) *mailer {
+	return &mailer{
+		messageQueue: messageQueue,
+		logger:       logger,
+	}
+}
+
+// BootstrapMailer setup the mailer, the connections and start the async message shipment
+func BootstrapMailer(ctx context.Context, conns []*genesis.TelemetryEndpoint, logger log.LeveledLogger) error {
+	const (
+		maxRetries = 5
+		retryDelay = time.Second * 15
```

Comment:
You could move this down right above `timer := time.NewTimer(retryDelay)` to make it even more local. Forgive me I'm a local-everything maniac. Also compilation wise it doesn't change since it's a constant.